src/d3-ext/profile.js

```diff
@@ -58,13 +62,13 @@ ngeo.profile = function() {
    * @type {function(Object)}
    * Hover callback function.
    */
-  var hoverCallback = goog.nullFunction;
+  var hoverCallback = options.hoverCallback || goog.nullFunction;
```

Comment:
Honestly, having `options.hoverCallback` repeated makes it harder to read and easy to make typos.

While both expressions are not synonyms, they are both doing their job (I do not think we should use
goog.isDefAndNotNull).

I disagree with you on the second point: I expect all developers working on Ngeo to understand basics concepts of javascript. Then it is just a matter of having read the contributor guide. Don't you think preventing developers to use basic javascript is a bit tyrannical? ;-)